code/iaas/lb-instance/src/main/java/io/cattle/platform/lb/instance/process/HostRemovePreAction.java

```diff
@@ -0,0 +1,52 @@
+package io.cattle.platform.lb.instance.process;
+
+import io.cattle.platform.core.constants.HostConstants;
+import io.cattle.platform.core.dao.GenericMapDao;
+import io.cattle.platform.core.model.Host;
+import io.cattle.platform.core.model.LoadBalancerHostMap;
+import io.cattle.platform.engine.handler.HandlerResult;
+import io.cattle.platform.engine.handler.ProcessPreListener;
+import io.cattle.platform.engine.process.ProcessInstance;
+import io.cattle.platform.engine.process.ProcessState;
+import io.cattle.platform.object.process.StandardProcess;
+import io.cattle.platform.process.common.handler.AbstractObjectProcessLogic;
+import io.cattle.platform.util.type.Priority;
+
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
+@Named
+public class HostRemovePreAction extends AbstractObjectProcessLogic implements ProcessPreListener, Priority {
```

Comment:
So we don't confuse ourselves, lets go with a name like "HostRemovePreListener"  so that the class name corresponds to the fact it's a ProcessPreListener.